src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpFormatter.cs

```diff
@@ -18,30 +18,11 @@
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Formatting;
 
-internal class CSharpFormatter
+internal sealed class CSharpFormatter(IRazorDocumentMappingService documentMappingService)
 {
     private const string MarkerId = "RazorMarker";
 
-    private readonly IRazorDocumentMappingService _documentMappingService;
-    private readonly IClientConnection _clientConnection;
-
-    public CSharpFormatter(
-        IRazorDocumentMappingService documentMappingService,
-        IClientConnection clientConnection)
-    {
-        if (documentMappingService is null)
-        {
-            throw new ArgumentNullException(nameof(documentMappingService));
-        }
-
-        if (clientConnection is null)
-        {
-            throw new ArgumentNullException(nameof(clientConnection));
-        }
-
-        _documentMappingService = documentMappingService;
-        _clientConnection = clientConnection;
-    }
+    private readonly IRazorDocumentMappingService _documentMappingService = documentMappingService;
```

Comment:
Since you're removing the `ArgumentNullException`, could `IRazorDocumentMappingService` ever be null?

Reply:
This class is directly constructed in code, so nullable reference types protects us from ever passing in null. If it was created by DI, then both of the DI technologies we use would throw errors during composition/container creation and never get this far anyway.

Reply:
The fact that both MEF and MS.Ext.DependencyInjection already protect against null is a good one to remember. We have lots of services and endpoints that validate against null arguments that are wholly unnecessary. As more and more code is nullable enabled (most of tooling is and most of the compiler is not), removing unnecessary ArgumentNullExceptions gets a thumbs up from me.